presto-parquet/src/test/java/com/facebook/presto/parquet/reader/TestData.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.parquet.reader;
+
+import com.facebook.presto.common.type.Decimals;
+import com.google.common.primitives.Longs;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.util.Random;
+import java.util.function.IntFunction;
+
+import static com.facebook.presto.parquet.batchreader.BytesUtils.propagateSignBit;
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.slice.SizeOf.SIZE_OF_LONG;
+import static java.lang.Math.toIntExact;
+
+public final class TestData
+{
+    private TestData() {}
+
+    // Based on org.apache.parquet.schema.Types.BasePrimitiveBuilder.maxPrecision to determine the max decimal precision supported by INT32/INT64
+    public static int maxPrecision(int numBytes)
+    {
+        return toIntExact(
+                // convert double to long
+                Math.round(
+                        // number of base-10 digits
+                        Math.floor(Math.log10(
+                                Math.pow(2, 8 * numBytes - 1) - 1))));  // max value stored in numBytes
+    }
+
+    public static IntFunction<long[]> unscaledRandomShortDecimalSupplier(int bitWidth, int precision)
+    {
+        long min = (-1 * Decimals.longTenToNth(precision)) + 1;
+        long max = Decimals.longTenToNth(precision) - 1;
+        return size -> {
+            Random random = new Random(1);
+            long[] result = new long[size];
+            for (int i = 0; i < size; i++) {
+                result[i] = Math.max(
+                        Math.min(randomLong(random, bitWidth), max),
```

Comment:
avoid random numbers in tests. Test results need to be reproducible.

Reply:
Fixed.